allennlp/models/model.py

```diff
@@ -365,9 +365,33 @@ def extend_embedder_vocab(self, embedding_sources_mapping: Dict[str, str] = None
                     self.vocab, extension_pretrained_file=pretrained_file, model_path=model_path
                 )
 
+    @classmethod
+    def from_archive(cls, archive_file: str, vocab: Vocabulary = None) -> "Model":
```

Comment:
Probably a dumb question, but how does this work from a config file without the `Lazy`  parameter? Is `Vocabulary` an exception to that rule?

Reply:
This does make use of `Lazy`, but it's [these `Lazy`s](https://github.com/allenai/allennlp/blob/eb9abcf370fed33dc0e47b8420c2f5b96de0de70/allennlp/commands/train.py#L532-L535) that get leveraged.  By the time we get to this class, the `Vocabulary` has already been constructed, so we don't need another `Lazy` annotation (for the same reason that `Model.__init__` also doesn't need `Lazy[Vocabulary]` - that gets passed in using the `**extras` stuff in `FromParams`, seen in this code in [this line](https://github.com/allenai/allennlp/blob/eb9abcf370fed33dc0e47b8420c2f5b96de0de70/allennlp/commands/train.py#L629)).  `Lazy` is only necessary if you have multiple arguments to a *single function* that have sequential dependencies, like a `Vocabulary` that has its own parameters but depends on training data for its construction.